sentence_transformers/SentenceTransformer.py

```diff
@@ -449,6 +476,7 @@ def save_to_hub(self,
         :param exist_ok: If true, saving to an existing repository is OK. If false, saving only to a new repository is possible
         :param replace_model_card: If true, replace an existing model card in the hub with the automatically created model card
         :param train_datasets: Datasets used to train the model. If set, the datasets will be added to the model card in the Hub.
+        :param local_repo_path: Local path where the model repo will be stored. If not provided, a temporary directory will be created.
```

Comment:
I'm not convinced by the behaviour of using a temporary directory by default. This would mean the repo is cloned every single save, which would make it very slow. Should it use something like "hub" by default maybe?

Reply:
I'm okay with either way! I think we'd decided to keep the temp directory by default from this convo: https://huggingface.slack.com/archives/C03HVLC16D7/p1657743607849189?thread_ts=1657694774.984229&cid=C03HVLC16D7

Reply:
Hmm good point :thinking: after thinking more about it I think having "hub" by default might make more sense to avoid too many redownloads